components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/authz/AuthorizationHandlerManagerTest.java

```diff
@@ -206,4 +207,44 @@ public void testHandleAuthorizationTokenResponseNoScopes() throws Exception {
         Assert.assertNotNull(respDTO.getAccessToken(), "Access token returned is null");
     }
 
+    @Test
+    public void testHandleAuthorizationTokenResponseUnauthorizedAccess() throws Exception {
+        authorizationHandlerManager = AuthorizationHandlerManager.getInstance();
+        authzReqDTO.setResponseType(TestConstants.AUTHORIZATION_HANDLER_RESPONSE_TYPE_TOKEN);
+        authzReqDTO.setConsumerKey(TestConstants.CLIENT_ID);
+        authzReqDTO.setScopes(TestConstants.SCOPE_UNAUTHORIZED_ACCESS.split(" "));
+        AuthenticatedUser user = new AuthenticatedUser();
+        user.setUserName(TestConstants.USER_NAME);
+        user.setTenantDomain(TestConstants.TENANT_DOMAIN);
+        user.setUserStoreDomain(TestConstants.USER_DOMAIN_PRIMARY);
+        authzReqDTO.setUser(user);
+        OAuth2AuthorizeRespDTO respDTO = authorizationHandlerManager.handleAuthorization(authzReqDTO);
+        String errorCode = respDTO.getErrorCode();
+        Assert.assertNotNull(respDTO, "Response is null");
+        Assert.assertNotNull(respDTO.getErrorCode(), "Error code returned is null");
+        Assert.assertEquals(errorCode, OAuthError.CodeResponse.UNAUTHORIZED_CLIENT,
+                            "Expected " + OAuthError.CodeResponse.UNAUTHORIZED_CLIENT + " error code but found : " +
```

Comment:
We can just say omit the string description, as the failure will anyway display the same text.